### PR TITLE
Refactor spacer block to share edit method with the web.

### PR DESCRIPTION
**Pull request by SergioEstevao**
## Description

Refactor the spacer block in order to have the same code for the edit method with web and mobile.

This involved refactoring the Block Controls on the web to use a standard component and implement a mobile version of the ResizableBox component. 

## How has this been tested?

This can be tested on this PR for GB-mobile:  https://github.com/wordpress-mobile/gutenberg-mobile/pull/1991

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/spacer/edit.js
Comment (ZebulanStanphill) [inline packages/block-library/src/spacer/edit.js]: Using a standard function declaration has the same effect while being clearer (it's more obviously a function), and being a few characters shorter.
```suggestion
	function changeAttribute( value ) {
		setAttributes( {
			height: value,
		} );
	}
```

Comment (Tug) [reply packages/block-library/src/spacer/edit.js]: @ZebulanStanphill I would tend to disagree here.
Using `function` or assigning an anonymous function to a `const` is not the same, the main difference being the time at which they are defined and the fact that a `const` cannot be changed. The `function` keyword basically works like `var` which is discouraged in the [WordPress Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#declaring-variables-with-const-and-let) for ESNext. That said there is nothing really explicit about it in the docs so we should use them at our convenance, remembering the differences between the two

File: packages/block-library/src/spacer/edit.js
Comment (Tug) [inline packages/block-library/src/spacer/edit.js]: `updateHeight` would be more explicit here

File: packages/block-library/src/spacer/edit.js
Comment (Tug) [inline packages/block-library/src/spacer/edit.js]: Seems like we're changing this component for the web as well, would be nice to have the approval from a core dev here

**Review by Tug — CHANGES_REQUESTED** [review]
Tested on mobile and it worked well 👍

**Comment by SergioEstevao** [comment]
@Tug updated the code according to your suggestions.

@ZebulanStanphill do you mind to give it another look?

File: packages/block-library/src/spacer/edit.js
Comment (ZebulanStanphill) [inline packages/block-library/src/spacer/edit.js]: The WordPress JavaScript Coding Standards [recommend using CAPS_WITH_UNDERSCORES for constants like these](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#constants).

File: packages/block-library/src/spacer/edit.js
Comment (ZebulanStanphill) [reply packages/block-library/src/spacer/edit.js]: Not a core dev, but this looks good to me. In my opinion, sliders should be used for controls where there is a large range of possible values and the exact value doesn't really matter that much. This seems like a perfect place to use a slider.

File: packages/block-library/src/spacer/edit.js
Comment (SergioEstevao) [reply packages/block-library/src/spacer/edit.js]: Updated the code to follow the recommendations.

**Comment by ZebulanStanphill** [comment]
Tested on desktop; everything seems to be working fine there. :+1:

**Review by ZebulanStanphill — APPROVED** [review]

**Review by Tug — APPROVED** [review]
LGTM 👍

**Pull request merged**
